src/mantarray_file_manager/files.py

```diff
@@ -118,6 +118,27 @@ def _extract_datetime_from_h5(
     open_h5_file: h5py._hl.files.File,  # pylint: disable=protected-access # WTF pylint...this is a type definition
     metadata_uuid: UUID,
 ) -> datetime.datetime:
+    if str(metadata_uuid) not in open_h5_file.attrs:
+        if metadata_uuid == UTC_BEGINNING_RECORDING_UUID:
+            # Tanner (9/17/20): The use of this proxy value is justified by the fact that there is a 15 second delay between when data is recorded and when the GUI displays it, and because the GUI will send the timestamp of when the recording button is pressed.
+            acquisition_timestamp_str = open_h5_file.attrs[
+                str(UTC_BEGINNING_DATA_ACQUISTION_UUID)
+            ]
+            begin_recording = datetime.datetime.strptime(
+                acquisition_timestamp_str, DATETIME_STR_FORMAT
+            ).replace(tzinfo=datetime.timezone.utc) + datetime.timedelta(seconds=15)
+            return begin_recording
+        if metadata_uuid == UTC_FIRST_TISSUE_DATA_POINT_UUID:
+            # Tanner (9/17/20): Early file versions did not include this metadata under a UUID, so we have to use this string identifier instead
+            metadata_name = "UTC Timestamp of Beginning of Recorded Tissue Sensor Data"
+            timestamp_str = open_h5_file.attrs[str(metadata_name)]
+            return datetime.datetime.strptime(
+                timestamp_str, DATETIME_STR_FORMAT
+            ).replace(tzinfo=datetime.timezone.utc)
+        raise NotImplementedError(
+            f"This UUID is not used in Mantarray Files: {str(metadata_uuid)}"
```

Comment:
I would be more general----"not used" wasn't what caused us to first find the error in the first place.  We were looking for a UUID in a file that didn't happen to have it.  I would have the error message say that the UUID was not found in THIS file, which is file format version x.y.z

Reply:
Also, this should be a tested error.  We need to reserve NotImplementedErrors for things where we have no idea how it could possibly happen---mostly to make mypy happy.  They are excluded from our code coverage because we never expect them to actually happen.   We definitely do expect this specific case to happen---it happened today, so it could definitely happen again in the future if we have a situation where a version of an H5 file is unintentionally not fully supported